core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataDeleteIndexService.java

```diff
@@ -99,6 +100,10 @@ public ClusterState execute(final ClusterState currentState) {
                     routingTableBuilder.remove(indexName);
                     clusterBlocksBuilder.removeIndexBlocks(indexName);
                     metaDataBuilder.remove(indexName);
+                    // create the index tombstone in the cluster state
+                    Optional<IndexGraveyard.Tombstone> maybePurged = metaDataBuilder.indexGraveyard().addTombstone(index);
```

Comment:
see comment about making purging explicit. Also we should add the purge method to MetaDataCreateIndexService as well?